cmd/dp-code-list-api/main.go

```diff
@@ -5,92 +5,155 @@ import (
+		shutdownCtx, cancel := context.WithTimeout(context.Background(), cfg.GracefulShutdownTimeout)
+		anyError := false
+		log.Event(shutdownCtx, "shutdown with timeout", log.INFO, log.Data{"timeout": cfg.GracefulShutdownTimeout})
 
-		if httpShutdown {
-			err = httpServer.Shutdown(ctx)
-			if err != nil {
-				log.Error(err, nil)
-			}
+		// StopHTTP Server
+		err = httpServer.Shutdown(shutdownCtx)
+		if err != nil {
+			anyError = true
+			log.Event(shutdownCtx, "http server shutdown error", log.ERROR, log.Error(err))
+		} else {
+			log.Event(shutdownCtx, "http server successful shutdown", log.INFO)
 		}
 
-		healthTicker.Close()
+		// Strop healthcheck
+		hc.Stop()
+		log.Event(shutdownCtx, "healthcheck stopped", log.INFO)
 
-		if err = datastore.Close(ctx); err != nil {
-			log.Error(err, nil)
+		// Close data store
+		if err = datastore.Close(shutdownCtx); err != nil {
+			anyError = true
+			log.Event(shutdownCtx, "datastore close error", log.ERROR, log.Error(err))
+		} else {
+			log.Event(shutdownCtx, "datastore successfully closed", log.INFO)
 		}
 
-		log.Info("shutdown complete", nil)
-
+		// cancel the timer in the shutdown context.
 		cancel()
-		os.Exit(1)
+
+		// if any error happened during shutdown, log it and exit with err code
+		if anyError {
+			log.Event(ctx, "graceful shutdown had errors", log.WARN)
+			os.Exit(1)
+		}
+
+		// if all dependencies shutted down successfully, log it and exit with success code
+		log.Event(ctx, "graceful shutdown was successful", log.INFO)
+		os.Exit(0)
 	}
 
+	// Start healthcheck ticker
+	hc.Start(ctx)
+
+	// Start HTTP Server
 	go func() {
-		log.Info("code list api starting.....", log.Data{"bind_addr": cfg.BindAddr})
+		log.Event(ctx, "code list api starting.....", log.INFO, log.Data{"bind_addr": cfg.BindAddr})
 		if err := httpServer.ListenAndServe(); err != nil && err != http.ErrServerClosed {
-			log.Error(err, nil)
+			hc.Stop()
+			log.Event(ctx, "error starting http server", log.ERROR, log.Error(err))
 			httpErrChannel <- err
 		} else {
 			httpErrChannel <- errors.New("http completed shutdown")
 		}
 	}()
 
-	for {
-		select {
-		case err := <-httpErrChannel:
-			log.ErrorC("http error received", err, nil)
-			shutdown(false)
-		case <-signals:
-			log.Debug("os signal received", nil)
-			shutdown(true)
+	// log errors
+	go func() {
+		for {
+			select {
+			case err := <-httpErrChannel:
+				log.Event(ctx, "http error received", log.ERROR, log.Error(err))
+			}
 		}
+	}()
+
+	// wait until we receive a signal
+	<-signals
+	log.Event(ctx, "os signal received", log.INFO)
+	shutdown()
+}
+
+// RegisterCheckers adds the checkers for the provided clients to the healthcheck object.
+func registerCheckers(ctx context.Context, hc *healthcheck.HealthCheck, db *graph.DB) (err error) {
+
+	hasErrors := false
+
+	if err = hc.AddCheck("Neo4J", db.Checker); err != nil {
+		hasErrors = true
+		log.Event(nil, "error adding check for graph db", log.ERROR, log.Error(err))
```

Comment:
context is passed into the log message as nil. This should be changed to use `ctx`